packages/entities/entities-plugins/src/components/PluginForm.vue

```diff
@@ -90,130 +147,687 @@ const props = defineProps({
+const record = ref<Record<string, any> | null>(null)
+const configResponse = ref<Record<string, any>>({})
+const formLoading = ref(false)
+const formFieldsOriginal = reactive<PluginFormFields>({
+  enabled: true,
+  protocols: [],
+  tags: [],
+})
 
 const form = reactive<PluginFormState>({
   fields: {
-    name: '',
-    tags: '',
-    entity_id: '',
+    enabled: true,
+    protocols: [],
+    tags: [],
   },
   isReadonly: false,
   errorMessage: '',
 })
 
-const formFieldsOriginal = reactive<PluginFormFields>({
-  name: '',
-  tags: '',
-  entity_id: '',
+// non-editable plugin type. They shouldn't be able to get to this unless they manually
+// type in the URL
+const isDisabled = computed((): boolean => {
+  const currentPlugin = Object.keys(customSchemas).find((key: string) => key === props.pluginType)
+
+  return currentPlugin ? (customSchemas[currentPlugin as keyof typeof customSchemas] as Record<string, any>)?.configurationDisabled : false
+})
+
+const entityData = computed((): PluginEntityInfo => {
+  const consumerId = (props.config.entityType === 'consumers' && props.config.entityId) || record.value?.consumer?.id
+  const consumerGroupId = (props.config.entityType === 'consumer_groups' && props.config.entityId) || record.value?.consumer_group?.id
+  const serviceId = (props.config.entityType === 'services' && props.config.entityId) || record.value?.service?.id
+  const routeId = (props.config.entityType === 'routes' && props.config.entityId) || record.value?.route?.id
+
+  let entity = PluginScope.GLOBAL
+  let endpoint = props.config.entityType || 'plugins'
+
+  if (consumerId) {
+    entity = PluginScope.CONSUMER
+    endpoint = 'consumers'
+  } if (consumerGroupId) {
+    entity = PluginScope.CONSUMER_GROUP
+    endpoint = 'consumer_groups'
+  } else if (routeId) {
+    entity = PluginScope.ROUTE
+    endpoint = 'routes'
+  } else if (serviceId) {
+    entity = PluginScope.SERVICE
+    endpoint = 'services'
+  }
+
+  return {
+    entity,
+    entityEndpoint: endpoint,
+    id: consumerId || consumerGroupId || serviceId || routeId,
+  }
+})
+
+// Configuration for globally shared fields
+const defaultFormSchema: DefaultPluginsSchemaRecord = reactive({
+  enabled: {
+    type: 'switch',
+    model: 'enabled',
+    label: t('plugins.form.fields.enabled.label'),
+    textOn: t('plugins.form.fields.enabled.on_text'),
+    textOff: t('plugins.form.fields.enabled.off_text'),
+    styleClasses: 'field-switch bottom-border hide-label',
+    default: true,
+  },
+  // this is a required field that the user cannot set, it's always the name of the plugin
+  // ex. 'acl'
+  name: {
+    default: props.pluginType,
+    type: 'input',
+    inputType: 'hidden',
+    styleClasses: 'd-none',
+  },
+  // plugin scoping
+  selectionGroup: {
+    type: props.hideScopeSelection || (formType.value === EntityBaseFormType.Create && props.config.entityId) ? 'foreign' : 'selectionGroup',
```

Comment:
The comment for `hideScopeSelection` says `Rather than using internal logic to determine whether or not to hide scope, force it`, so could we make `hideScopeSelection` take precedence over internal checking?
```suggestion
    type: !props.hideScopeSelection ? 'selectionGroup' : ((formType.value === EntityBaseFormType.Create && props.config.entityId) ? 'foreign' : 'selectionGroup'),
```